core/server/suspend_test.go

```diff
@@ -105,14 +106,15 @@ func TestSuspend_Suspend(t *testing.T) {
 			Suspend: false,
 		}
 
-		md := metadata.Pairs(MetadataUserKey, "anne", MetadataGroupsKey, "system:masters")
+		principalID := "anne"
+		md := metadata.Pairs(MetadataUserKey, principalID, MetadataGroupsKey, "system:masters")
 		outgoingCtx := metadata.NewOutgoingContext(ctx, md)
 		_, err = c.ToggleSuspendResource(outgoingCtx, req)
 		g.Expect(err).NotTo(HaveOccurred())
 
 		for _, tt := range tests {
 			name := types.NamespacedName{Name: tt.obj.GetName(), Namespace: ns.Name}
-			g.Expect(checkSpec(t, k, name, tt.obj)).To(BeFalse())
+			g.Expect(checkSpec(t, k, principalID, name, tt.obj, false)).To(BeFalse())
```

Comment:
I'd probably have split this into to two expectations rather than piggy-backing on top of `checkSpec` ?

Reply:
I don't think you need to add a switch?

You've got a `client.Object` https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Object  which embeds https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object ?